src/io/free_functions.rs

```diff
@@ -102,13 +107,13 @@ pub(crate) fn image_dimensions_with_format_impl<R: BufRead + Seek>(fin: R, forma
         #[cfg(feature = "jpeg")]
         image::ImageFormat::Jpeg => jpeg::JpegDecoder::new(fin)?.dimensions(),
```

Comment:
These match arms ignore the limits. That seems contradictory. We might think about ignoring defaulted limits but if a limit has been set *explicitely* then this should not happen and I'd rather outright refuse to decode than ignore them.

Reply:
I see where you are coming from, but currently only three formats support limits. I think we should think a bit about this first.

Reply:
Actually, by incorporating @fintelia's suggestion with a `ImageDecoder::set_working_bytes` method this could be implemented quite elegantly.

Reply:
I'm aware that only three format support limits and we have only fuzzed png to ensure that the support is actually somewhat reasonable. But that's precisely my point, setting limits might be used as a level of sandboxing and it is completely unfit for that usecase when we then ignore the limits.